packages/file_selector/file_selector_web/lib/src/dom_helper.dart

```diff
@@ -52,6 +52,11 @@ class DomHelper {
       completer.completeError(platformException);
     });
 
+    inputElement.addEventListener('cancel', (Event event) {
+      inputElement.remove();
+      completer.complete(<XFile>[]);
+    });
```

Comment:
I don't think this code is implementing all the differences described here?

https://github.com/flutter/packages/pull/3683#issuecomment-1507825179

Reply:
I tried testing all scenarios, and was working as expected

Reply:
@raju8000 what browsers did you use to test? I want to take a look too! Thanks!

Reply:
@ditman  Current stable version of Firefox,  Chrome Canary & Safari Technology Preview for macOS Monterey